src/engraving/layout/v0/measurelayout.cpp

```diff
@@ -1669,7 +1669,7 @@ void MeasureLayout::addSystemHeader(Measure* m, bool isFirstSystem, LayoutContex
         } else if (kSegment && !isPitchedStaff) {
             EngravingItem* e = kSegment->element(track);
             if (e && e->isKeySig()) {
-                kSegment->remove(e);
+                kSegment->setVisible(false);
```

Comment:
I'm afraid this is not the correct solution: setting a segment invisible doesn't do anything in Musescore. In fact, at this point you may just as well remove the last `if else` block and it will produce the same result. I think the correct way to do this would be to delete the `KeySig` (i.e. the graphical item), but maintain the `KeySigEvent` (i.e. the object which holds the actual key information). I'll into this more and come back to you :)

Reply:
Or maybe use `setSkipDraw(true)`? We do that in some other situations where elements must not be rendered on TAB staves.

However, I just realise that this property does not do anything about preventing interaction. I'm afraid that what happens when the user tries to interact with such an element is undefined.

Reply:
Ok so: at the moment we avoid drawing key signatures on TAB staves thanks to [this check](https://github.com/musescore/MuseScore/blob/master/src/engraving/layout/v0/tlayout.cpp#L2898-L2900C6) which happens at layout stage. It it feels a bit hacky to me, but it is simple and helps with these remove/recreate issues, so I think we can work with it, for now. As for your solution @mikekirin, you can go ahead and delete the last `else if` block: it is anyway equivalent to what you already did :)